Tasks/DotNetCoreInstallerV1/versionfetcher.ts

```diff
@@ -0,0 +1,317 @@
+import { HttpClientResponse } from 'typed-rest-client/HttpClient';
+
+import * as utils from "./versionutilities";
+
+export class DotNetCoreVersionFetcher {
+    constructor() {
+        let proxyUrl: string = tl.getVariable("agent.proxyurl");
+        var requestOptions: httpInterfaces.IRequestOptions = proxyUrl ? {
+            proxy: {
+                proxyUrl: proxyUrl,
+                proxyUsername: tl.getVariable("agent.proxyusername"),
+                proxyPassword: tl.getVariable("agent.proxypassword"),
+                proxyBypassHosts: tl.getVariable("agent.proxybypasslist") ? JSON.parse(tl.getVariable("agent.proxybypasslist")) : null
+            }
+        } : {};
+
+        this.httpCallbackClient = new httpClient.HttpClient(tl.getVariable("AZURE_HTTP_USER_AGENT"), null, requestOptions);
+        this.channels = [];
+    }
+
+    public async getVersionInfo(versionSpec: string, packageType: string, includePreviewVersions: boolean): Promise<VersionInfo> {
+        var requiredVersionInfo: VersionInfo = null;
+        if (!this.channels || this.channels.length < 1) {
+            await this.setReleasesIndex();
+        }
+
+        let channelInformation = this.getVersionChannel(versionSpec);
+        if (channelInformation) {
+            requiredVersionInfo = await this.getVersionFromChannel(channelInformation, versionSpec, packageType, includePreviewVersions);
+        }
+
+        if (!requiredVersionInfo && !versionSpec.endsWith("x")) {
+            console.log(tl.loc("FallingBackToAdjacentChannels", versionSpec));
+            requiredVersionInfo = await this.getVersionFromOtherChannels(versionSpec, packageType, includePreviewVersions);
+        }
+
+        if (!requiredVersionInfo) {
+            throw tl.loc("VersionNotFound", packageType, versionSpec);
+        }
+
+        let dotNetSdkVersionTelemetry = `{"userVersion":"${versionSpec}", "resolvedVersion":"${requiredVersionInfo.version}"}`;
+        console.log("##vso[telemetry.publish area=TaskDeploymentMethod;feature=DotNetCoreInstallerV1]" + dotNetSdkVersionTelemetry);
+        return requiredVersionInfo;
+    }
+
+    public getDownloadUrl(versionInfo: VersionInfo, packageType: string): string {
+        console.log(tl.loc("GettingDownloadUrl", packageType, versionInfo.version));
+
+        let osSuffixes = this.detectMachineOS();
+        let downloadPackageInfoObject: VersionFilesData = null;
+        osSuffixes.find((osSuffix) => {
+            downloadPackageInfoObject = versionInfo.files.find((downloadPackageInfo: VersionFilesData) => {
+                if (downloadPackageInfo.rid && osSuffix && downloadPackageInfo.rid.toLowerCase() == osSuffix.toLowerCase()) {
+                    if ((osSuffix.split("-")[0] != "win" && osSuffix.split("-")[0] != "osx") || (osSuffix.split("-")[0] == "win" && downloadPackageInfo.name.endsWith(".zip")) || (osSuffix.split("-")[0] == "osx" && downloadPackageInfo.name.endsWith("tar.gz"))) {
```

Comment:
what does this mean? DOes this mean for linux we assume the entry will always be correct
 During extract, aren't we assuming tar gz for everything non-windows

Reply:
I was assuming so, as tar.gz is the only type of file available in the repo for linux, whereas mac had pkg and tar.gz
Also did not want to put restriction if tomorrow they use a different archive extension for linux.
@sachinma do you think we should do similar for linux ? or make != pkg condition for osx ?